control-plane/cni/main_test.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/consul/sdk/iptables"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestWaitForAnnotation(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name       string
+		annotation string
+		pod        func(*corev1.Pod) *corev1.Pod
```

Comment:
This might be clearer as `annotatePod`.

```suggestion
		annotatePod        func(*corev1.Pod) *corev1.Pod
```